lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -3093,6 +3093,70 @@ LogicalResult ConvertAtenOp<AtenBroadcastToOp>::matchAndRewrite(
       "unimplemented: broadcasts other than same rank or zero ranked tensor.");
 }
 
+template <>
+LogicalResult ConvertAtenOp<AtenGatherOp>::matchAndRewrite(
+    AtenGatherOp op, OpAdaptor adaptor,
+    ConversionPatternRewriter &rewriter) const {
+  // For easy understanding of this algorithm, I will comment the code with an exact example:
+  // torch.aten.gather (!torch.vtensor<[1,4,3],f32>, !torch.int-1, !torch.vtensor<[1,4,2],si64>)
+  // -> !torch.vtensor<[1,4,2],f32>
+  // https://gist.github.com/AmosLewis/2f18434397025211da4491735bcc6db6
+
+  // Not a tensor type.
+  auto selfType = adaptor.getSelf().getType().template dyn_cast<TensorType>();
+  if (!selfType)
+    return rewriter.notifyMatchFailure(
+        op, "Only tensor types input are currently supported");
+  int paramsRank = selfType.getShape().size();
+  if(paramsRank!=3){
+    // most of the model input is rank==3, axis==-1
+    return rewriter.notifyMatchFailure(
+        op, "Only rank==3 tensor input are currently supported,"
+            "TODO generalize convertTorchIndexToTfIndices with more rank. ");
+  }
+
+  // get the dim
+  int64_t dim{0};
+  if (!matchPattern(op.getDim(), m_TorchConstantInt(&dim)))
+    return rewriter.notifyMatchFailure(
+        op, "unimplemented: value `dim` should be a torch constant int");
+  int32_t axis{0};
+  if(dim >= 0) {
+    axis = dim;
+  }else{
+    axis = paramsRank -1;
+  }
+  if(axis != -1 && axis != 2){
```

Comment:
nit: run `git clang-format` to make sure style matches with rest of codebase

Reply:
Done